src/Games/NexusMods.Games.FOMOD/CoreDelegates/PresetGuidedInstaller.cs

```diff
@@ -0,0 +1,50 @@
+using System.Diagnostics;
+using NexusMods.Abstractions.Activities;
+using NexusMods.Abstractions.GuidedInstallers;
+
+namespace NexusMods.Games.FOMOD.CoreDelegates;
+
+/// <summary>
+/// A IGuidedInstaller implementation that uses a preset list of steps to make the same choices
+/// a user previously made for specific steps.
+/// </summary>
+public class PresetGuidedInstaller : IGuidedInstaller
+{
+    private readonly FomodOption[] _steps;
+    private int _currentStep = 0;
+
+    public PresetGuidedInstaller(FomodOption[] steps)
+    {
+        _steps = steps;
+    }
+    
+    public void Dispose()
+    {
+    }
+
+    public void SetupInstaller(string name)
+    {
+    }
+
+    public void CleanupInstaller()
+    {
+    }
+
+    public Task<UserChoice> RequestUserChoice(GuidedInstallationStep installationStep, Percent progress, CancellationToken cancellationToken)
+    {
+        var step = _steps[_currentStep];
+        
+        // This looks gross, but it's fairly simple we map through the two trees matching by name, and it's cleaner than 4 nested loops.
+        var choices = 
```

Comment:
Do we know if the names are case sensitive? I imagine the preset choices aren't set by humans, but if they can be modified by humans, a case insensitive comparison may be better here.